File: Workspace Eclipse/InsertionSort/insertion_sort.c
#include <stdio.h>
#include <time.h>
#include "vector.h"

void insertion_sort(vector* vector){

	for(int i=1; i< vector->dim; i++){
		int key = vector->v[i];

		int j = i-1;
		while(j>= 0 && vector->v[j]>key){
			vector->v[j+1] = vector->v[j];
			j=j-1;
		}
		vector->v[j+1] = key;
	}

}

void insertion_sort_f(double* vector, int dimension){

	for(int i=1; i< dimension; i++){
		float key = vector[i];
		//printf("\n\n %f \n\n",vector[i]);
		int j = i-1;
		while(j>= 0 && vector[j]>key){
			vector[j+1] = vector[j];
			j=j-1;
		}
		vector[j+1] = key;
		//printf("\n\n %f \n\n",vector[j+1]);
	}

}

File: Workspace Eclipse/Insertion Sort/src/main_package/InsertionSort.java
package main_package;

public class InsertionSort{

	//private static final int N = 500;
	
	public static Vector insertionSort(Vector v) {
		
		for(int i=1; i< v.length(); i++) {
			int key = v.get(i);
			
			int j = i-1;
			while(j>=0 && v.get(j)>key) {
				v.set(j+1, v.get(j));
				j--;
			}
			v.set(j+1, key);
		}
		return v;
	}
	
	public static int[] insertionSort(int[] v) {
		
		for(int i=1; i< v.length; i++) {
			int key = v[i];
			
			int j = i-1;
			while(j>=0 && v[j]>key) {
				v[j+1] = v[j];
				j--;
			}
			v[j+1] =key;
		}
		return v;
	}

}
